module/purpose/ScriptTuner/src/ScriptTuner.cpp

```diff
@@ -141,25 +158,114 @@ namespace module::purpose {
             emit(std::move(waypoint));
         });
 
+        // Debugging to make sure logs are working
+        /// TODO: Remove before merging with main
+        on<Every<5, std::chrono::seconds>>().then("Heartbeat", [this] {
+            static size_t counter = 0;
+            log<NUClear::DEBUG>("Beep Boop ", counter++);
+        });
+
+        on<Every<AUTOSAVE_INTERVAL, std::chrono::seconds>, Single>().then("Autosave", [this] {
+            log<NUClear::DEBUG>("Autosaving script if required...");
+
+            // Only autosave if autosaving is enabled and we have unsaved changes
+            if (!autosave_enabled || !unsaved_changes)
+                return;
+
+            // Before we start, save the temp save the old path so we can delete it
+            // once we have a new autosave file.
+            auto last_autosave_path = this->autosave_path;
+
+            // Split the script name into the path and filename
+            auto [autosave_dir, filename] = utility::file::pathSplit(script_path);
+
+            // Pick the right autosave directory based on the configured save location
+            if (autosave_on_robot) {
+                // If we're autosaving on the robot, then save to an `autosave/` subdirectory
+                // in the same folder of the current script.
+                autosave_dir += "/autosave/";
+            }
+            else {
+                // Otherwise, if we're running in the docker container, then replace the
+                // autosave directory with a subdirectory in the mounted `../NUbots/` folder.
+                autosave_dir = "/home/nubots/NUbots/.scripttuner/autosave/";
+            }
```

Comment:
I feel like it would be better to have the config variable be the directory rather than a bool and then hardcoding the respective paths

Reply:
done